ms2deepscore/data_generators.py

```diff
@@ -10,150 +11,188 @@
 
 class DataGeneratorAllInchikeys(Sequence):
     """Generates data for training a siamese Keras model
+
+    This generator will provide training data by picking each training InchiKey
+    listed in *inchikey_ids* num_turns times in every epoch. It will then randomly
+    pick one the spectra corresponding to this InchiKey (if multiple) and pair it
+    with a randomly chosen other spectrum that corresponds to a reference score
+    as defined in same_prob_bins.
     """
-    def __init__(self, spectrums_binned_dicts: list, training_ids: list,
-                 y_true: np.ndarray, inchikey_mapping: pd.DataFrame, batch_size: int = 32,
-                 num_turns: int = 1,
+    def __init__(self, spectrums_binned: List[dict], score_array: np.ndarray,
+                 inchikey_ids: list, inchikey_score_mapping: np.ndarray,
+                 inchikeys_all: np.ndarray, dim: int,
+                 batch_size: int = 32, num_turns: int = 1,
                  peak_scaling: float = 0.5,
-                 dim: tuple = (10000,1), shuffle: bool = True, ignore_equal_pairs: bool = True,
-                 inchikeys_array: np.ndarray = None,
+                 shuffle: bool = True, ignore_equal_pairs: bool = True,
                  same_prob_bins: list = [(0, 0.5), (0.5, 1)],
-                 augment_peak_removal: dict = {"max_removal": 0.2, "max_intensity": 0.2},
+                 augment_peak_removal_max: float = 0.2,
+                 augment_peak_removal_intensity: float = 0.2,
                  augment_intensity: float = 0.1):
         """Generates data for training a siamese Keras model.
-        This generator will provide training data by picking each training InchiKey
-        listed in *training_ids* num_turns times in every epoch. It will then randomly
-        pick one the spectra corresponding to this InchiKey (if multiple) and pair it
-        with a randomly chosen other spectrum that corresponds to a reference score
-        as defined in same_prob_bins.
 
         Parameters
         ----------
-        spectrums_binned_dicts
+        spectrums_binned
             TODO: Make this less nested and clearer what I should exactly pass.
             List of dictionaries with the binned peak positions and intensities.
-        training_ids
-            List of IDs from the spectrums_binned_dicts list to use for training
-        y_true
-            TODO: In same order as what?
-            Array of reference similarity scores (=labels).
-        inchikey_mapping
-            TODO: Needs documentation
+        score_array
+            2D-array of reference similarity scores (=labels).
+            Must be symmetric (score_array[i,j] == score_array[j,i]) and in the
+            same order as inchikey_score_mapping.
+        inchikey_ids
+            List of IDs from unique_inchikeys to use for training.
```

Comment:
`unique_inchickeys` is `inchikey_score_mapping`?

Reply:
good catch. thanks!